temporal/api/cloud/cloudservice/v1/request_response.proto

```diff
@@ -845,3 +846,58 @@ message ValidateNamespaceExportSinkRequest {
 
 message ValidateNamespaceExportSinkResponse {
 }
+
+message CreateConnectivityRuleRequest {
+    temporal.api.cloud.connectivityrule.v1.ConnectivityRuleSpec spec = 1;
+    // The id to use for this async operation.
+    // Optional, if not provided a random id will be generated.
+    string async_operation_id = 2;
+}
+
+message CreateConnectivityRuleResponse {
+    string connectivity_rule_id = 1;
+    // The async operation
+    temporal.api.cloud.operation.v1.AsyncOperation async_operation = 2;
+}
+
+message GetConnectivityRuleRequest {
+    string connectivity_rule_id = 1;
+}
+
+message GetConnectivityRuleResponse {
+    temporal.api.cloud.connectivityrule.v1.ConnectivityRule connectivity_rule = 1;
+}
+
+message GetConnectivityRulesRequest {
+    // The requested size of the page to retrieve.
+    // Optional, defaults to 100.
+    int32 page_size = 1;
+    // The page token if this is continuing from another response.
+    // Optional, defaults to empty.
+    string page_token = 2;
+    // Namespace id
+    string namespace = 3;
+}
+
+message GetConnectivityRulesResponse {
+    // connectivity_rules returned
+    repeated temporal.api.cloud.connectivityrule.v1.ConnectivityRule connectivity_rules = 1;
+    // The next page token
+    string next_page_token = 2;
+}
+
+message DeleteConnectivityRuleRequest {
+    // The ID of the connectivity rule that need be deleted, required.
+    string connectivity_rule_id = 1;
+
+    // The resource version which should be the same from the the db, required
```

Comment:
Instead of talking about the DB callers should get this value from the GetConnectivityRule API.